### DISCUSSION TITLE
Replace Cell Values at Each Z-Height

### DISCUSSION THREAD

**Discussion by adamgranthendry** [pyvista/pyvista#1582] [Q&A] [answered]
@akaszynski What's the simplest way to replace specific surface mesh cells at a single z-slice with a constant value. I'm not looking to map by height, but rather for example, for the `hills_data`, can all the "Elevation" values for cells on the surface at a certain height be replaced with a constant value?

**Comment by adamgranthendry** [comment]
@akaszynski @banesullivan I can get the points of the mesh with `.points`, but I need to extract those cells of the mesh between two z-values and replace the `.point_array` data for those cells with the constant value. How can I achieve this? Can I get the corresponding cell indices by this?:

```python
z0 = 0.0
z1 = 1.0
nds = np.argwhere(mesh.points[:,2] >= z0 & mesh.points[:,2] < z1)
```

**Comment by akaszynski** [comment, marked as answer]
I'd do it like this:

```py
z0 = 0.0
z1 = 1.0
mask = np.logical_and(mesh.points[:,2] >= z0, mesh.points[:,2] < z1)
my_mesh.points_array[mask] = constant_value
```
